core/chains/evm/logpoller/orm.go

```diff
@@ -129,3 +129,83 @@ func (o *ORM) SelectLogsByBlockRangeFilter(start, end int64, address common.Addr
 	}
 	return logs, nil
 }
+
+// LatestLogEventSigsAddrsWithConfs finds the latest logs by block that matches a list of addresses and list of events
+func (o *ORM) LatestLogEventSigsAddrsWithConfs(fromBlock int64, addresses []common.Address, eventSigs []common.Hash, confs int, qopts ...pg.QOpt) ([]Log, error) {
+	var logs []Log
+
+	var latestBlocks []int64
+
+	arg := map[string]interface{}{
+		"addresses": addresses,
+		"sigs":      eventSigs,
+		"confs":     confs,
+		"chainid":   utils.NewBig(o.chainID),
+		"fromBlock": fromBlock,
+	}
+
+	// Get the latest block with a matching update for each address
+	query, args, err := sqlx.Named(`
+	SELECT MAX(block_number) as block_number FROM logs
+	WHERE evm_chain_id = :chainid
+	AND address IN (:addresses)
+	AND event_sig IN (:sigs)
+	AND block_number > :fromBlock
+	AND (block_number + :confs) <= (
+		SELECT COALESCE(block_number, 0)
+		FROM log_poller_blocks
+		WHERE evm_chain_id = :chainid
+		ORDER BY block_number DESC
+		LIMIT 1
+	)
+	GROUP BY address, event_sig, evm_chain_id
+	ORDER BY block_number DESC`,
+		arg,
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	query, args, err = sqlx.In(query, args...)
+	if err != nil {
+		return nil, err
+	}
+
+	q := o.q.WithOpts(qopts...)
+	query = o.q.Rebind(query)
+	err = q.Select(&latestBlocks, query, args...)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(latestBlocks) == 0 {
+		return logs, nil
+	}
+
+	arg["latestblocks"] = latestBlocks
+
+	query, args, err = sqlx.Named(`
+		SELECT * FROM logs 
+		WHERE evm_chain_id = :chainid 
+        AND block_number IN (:latestblocks)
+		ORDER BY (block_number, log_index) DESC`,
+		arg,
```

Comment:
I'm curious, does this leave an ever-growing number of unpruneable logs in the database that will never be consumed?

Reply:
I guess that might be a qq for @connorwstein ? fwdMgr only care about latest logs that come after node boot.

Reply:
correct - there is no pruning yet for the log poller, ticket is in the backlog. Currently it grows indefinitely in the same same manner as the existing log_broadcasts table